src/main/java/com/chrosciu/bootcamp/tasks/github/GithubClient.java

```diff
@@ -4,28 +4,33 @@
 import com.chrosciu.bootcamp.tasks.github.dto.Repository;
 import lombok.RequiredArgsConstructor;
 import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
 
 @RequiredArgsConstructor
 public class GithubClient {
     private final GithubApi githubApi;
 
     public Flux<Repository> getUserRepositories(String username) {
-        //TODO: Implement
-        return null;
+        Mono<List<Repository>> results = githubApi.getUserRepositories(username);
+        return results.flatMapMany(Flux::fromIterable);
     }
 
     public Flux<Branch> getUserRepositoryBranches(String username, String repo) {
-        //TODO: Implement
-        return null;
+        Mono<List<Branch>> branches = githubApi.getUserRepositoryBranches(username, repo);
+        return branches.flatMapMany(Flux::fromIterable);
     }
 
     public Flux<Repository> getUsersRepositories(Flux<String> usernames) {
-        //TODO: Implement
-        return null;
+        return usernames.flatMap(this::getUserRepositories);
     }
 
     public Flux<String> getAllUserBranchesNames(String username) {
-        //TODO: Implement
-        return null;
+        return getUsersRepositories(Flux.just("SebastianRoslon"))
+                .flatMap(r -> getUserRepositoryBranches(username, r.getName()))
+                .distinct()
```

Comment:
To dziala, ale troche z przypadku - dlatego ze klasa Branch ma tylko jedno pole bedace wlasnie nazwa brancha, natomiast gdyby pojawilo sie jakiekolwiek inne to ten kod przestanie dzialac :) Czy masz pomysl jak to naprawic ?